frontend/src/component/layout/MainLayout/NavigationSidebar/useRoutes.ts

```diff
@@ -9,41 +9,57 @@ import {
 import { useInstanceStatus } from 'hooks/api/getters/useInstanceStatus/useInstanceStatus';
 import type { INavigationMenuItem } from 'interfaces/route';
 import type { IUiConfig } from 'interfaces/uiConfig';
+import { useUiFlag } from 'hooks/useUiFlag';
+import { useVariant } from 'hooks/useVariant';
+import type { Variant } from 'utils/variants';
 
-const filterRoutes = (
-    routes: INavigationMenuItem[],
-    uiConfig: IUiConfig,
-    { enterprise, pro, billing }: PlanData,
-) => {
-    return routes
-        .filter(filterByConfig(uiConfig))
-        .filter((route) =>
-            filterRoutesByPlanData(route?.menu, {
-                enterprise,
-                pro,
-                billing,
-            }),
-        )
-        .map(normalizeRoutePath);
+const useNewRoute = () => {
+    const flag = useUiFlag('newInUnleash');
+    return useVariant(flag as Variant);
 };
 
+const markRouteIfNew = (
+    route: INavigationMenuItem,
+    newRouteTitle?: string,
+): INavigationMenuItem => {
+    if (newRouteTitle?.trim().toLowerCase() === route.title.toLowerCase()) {
+        return { ...route, isNew: true };
+    }
+    return route;
+};
+
+const filterMapRoutes =
+    (uiConfig: IUiConfig, planData: PlanData, newRouteTitle?: string) =>
+    (routes: INavigationMenuItem[]) => {
+        return routes
+            .filter(filterByConfig(uiConfig))
+            .filter((route) => filterRoutesByPlanData(route?.menu, planData))
+            .map((route) => {
+                const normalized = normalizeRoutePath(route);
+                return markRouteIfNew(normalized, newRouteTitle);
```

Comment:
This is really the only change to this function: after normalizing, add the `isNew` marker if it's new.